pennylane/ops/qutrit/non_parametric_ops.py

```diff
@@ -0,0 +1,200 @@
+import numpy as np
+
+import pennylane as qml  # pylint: disable=unused-import
+from pennylane.operation import Operation
+
+OMEGA = np.exp(2 * np.pi * 1j / 3)
+ZETA = OMEGA ** (1 / 3)  # ZETA will be used as a phase for later non-parametric operations
+
+# Note: When Operation.matrix() is used for qutrit operations, `wire_order` must be `None` as specifying
+# an order that is expected to return a permuted and expanded matrix different from the canonical matrix
+# will lead to errors as `expand_matrix()` in `pennylane/operation.py`, which is used to compute the
+# permuted and expanded matrix from the canonical matrix, is hard coded to work correctly for qubits.
+
+
+class TShift(Operation):
+    r"""TShift(wires)
+    The qutrit shift operator
+
+    .. math:: TShift = \begin{bmatrix}
+                        0 & 0 & 1 \\
+                        1 & 0 & 0 \\
+                        0 & 1 & 0
+                    \end{bmatrix}
+
+    **Details:**
+
+    * Number of wires: 1
+    * Number of parameters: 0
+
+    Args:
+        wires (Sequence[int] or int): the wire the operation acts on
+    """
+    num_wires = 1
+    """int: Number of wires that the operator acts on."""
+
+    num_params = 0
+    """int: Number of trainable parameters that the operator depends on."""
+
+    @staticmethod
+    def compute_matrix():
+        r"""Representation of the operator as a canonical matrix in the computational basis (static method).
+
+        The canonical matrix is the textbook matrix representation that does not consider wires.
+        Implicitly, this assumes that the wires of the operator correspond to the global wire order.
+
+        .. seealso:: :meth:`~.TShift.matrix`
+
+        Returns:
+            ndarray: matrix
+
+        **Example**
+
+        >>> print(qml.TShift.compute_matrix())
+        [[0 0 1]
+         [1 0 0]
+         [0 1 0]]
+        """
+        return np.array([[0, 0, 1], [1, 0, 0], [0, 1, 0]])
+
+    @staticmethod
+    def compute_eigvals():
+        r"""Eigenvalues of the operator in the computational basis (static method).
+
+        If :attr:`diagonalizing_gates` are specified and implement a unitary :math:`U`,
+        the operator can be reconstructed as
+
+        .. math:: O = U \Sigma U^{\dagger},
+
+        where :math:`\Sigma` is the diagonal matrix containing the eigenvalues.
+
+        Otherwise, no particular order for the eigenvalues is guaranteed.
+
+        .. seealso:: :meth:`~.TShift.eigvals`
+
+        Returns:
+            array: eigenvalues
+
+        **Example**
+
+        >>> print(qml.TShift.compute_eigvals())
+        [ -0.5+0.8660254j -0.5-0.8660254j 1. +0.j         ]
+        """
+        return np.array([OMEGA, OMEGA**2, 1])
+
+    # TODO: Add compute_decomposition once parametric ops are added.
+
+    def pow(self, z):
+        if isinstance(z, int):
+            z_mod3 = z % 3
+            if z_mod3 < 2:
+                return super().pow(z_mod3)
+            return [self.adjoint()]
```

Comment:
Is there a reason that these are treated separately? I guess it saves a matrix multiplication?

Reply:
Yes. Both `TShift` and `TClock` are periodic with period 3 when exponentiated, so this saves quite a lot of compute time if for some reason either of those operations are being raised to high integer powers

Reply:
Making a copy isn't necessary as the `adjoint` method returns a new object rather than setting altering its own `inverse` attribute.